core/src/window_service.rs

```diff
@@ -271,7 +273,19 @@ where
                                 };
                                 if shred_filter(&shred, last_root) {
                                     packet.meta.slot = shred.slot();
-                                    packet.meta.seed = shred.seed();
+
+                                    let bank = &bank_forks.read().unwrap()[shred.slot()];
+                                    let leader_pubkey = leader_schedule_cache.slot_leader_at(shred.slot(), Some(&bank));
+
+                                    /* TODO jbiseda
+                                    packet.meta.seed = if bank.feature_set.is_active(&feature_set::deterministic_shred_seed_enabled::id()) {
+                                        shred.deterministic_seed(leader_pubkey)
+                                    } else {
+                                        shred.deprecated_seed()
+                                    };
+                                    */
+                                    packet.meta.seed = shred.deterministic_seed(leader_pubkey);
+
```

Comment:
This logic is repeated both here and in `broadcast_stage.rs`. We probably better to encapsulate all these logic in one place.